core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -103,37 +103,8 @@ trait Runtime[+R] {
    * This method is effectful and should only be invoked at the edges of your program.
    */
   final def unsafeRunAsyncCancelable[E, A](zio: => ZIO[R, E, A])(k: Exit[E, A] => Any): Fiber.Id => Exit[E, A] = {
-    val InitialInterruptStatus = InterruptStatus.Interruptible
-
-    val fiberId = Fiber.newFiberId()
-
-    val scope = ZScope.unsafeMake[Exit[E, A]]()
-
-    val supervisor = platform.supervisor
-
-    lazy val context: FiberContext[E, A] = new FiberContext[E, A](
-      fiberId,
-      platform,
-      environment.asInstanceOf[AnyRef],
-      platform.executor,
-      InitialInterruptStatus,
-      None,
-      PlatformConstants.tracingSupported,
-      Platform.newWeakHashMap(),
-      supervisor,
-      scope
-    )
-
-    if (supervisor ne Supervisor.none) {
-      supervisor.unsafeOnStart(environment, zio, None, context)
-
-      context.onDone(exit => supervisor.unsafeOnEnd(exit, context))
-    }
-
-    context.evaluateNow(ZIOFn.recordStackTrace(() => zio)(zio.asInstanceOf[IO[E, A]]))
-    context.runAsync(k)
-
-    fiberId => unsafeRun(context.interruptAs(fiberId))
+    lazy val curZIO = if (Platform.isJS) zio else ZIO.yieldNow *> zio
```

Comment:
isn't the situation of JavaScript here similar to that of Native?

Reply:
I'm not sure what the situation is on Native here. At this point we have no tests for Native and there is an ongoing effort to get those working with associated changes so my overall point of view is we address Native specific issues as part of that but could definitely flip this to only yield on the JVM.